docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,484 @@
+   from composer.models import ComposerResNet
+   from composer.optim.scheduler import cosine_annealing_scheduler
+   from torch.optim import SGD
+
+   model = ComposerResNet(model_name="resnet50", num_classes=1000)
+   optimizer = SGD(model.parameters(), lr=0.1)
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='90ep',
+                     optimizer=optimizer,
+                     scheduler=consine_annealing_scheduler)
+
+.. seealso::
+
+    :doc:`Schedulers`<trainer/optimizers_and_schedulers>
+
+Training on GPU
+~~~~~~~~~~~~~~~
+
+Control which device you use for training with the ``device`` parameter,
+and we will handle the data movement and other systems-related
+engineering. We currently support the ``cpu`` and ``gpu`` devices.
+
+.. code:: python
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu')
+
+Distributed Training
+~~~~~~~~~~~~~~~~~~~~
+
+It's also simple to do data-parallel training on multiple GPUs. Composer
+provides a launcher command that works with the trainer and handles all
+the ``torch.distributed`` setup for you.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                                       device='gpu')
+   trainer.fit()
+
+Access the Composer launcher via the ``composer`` command along with the
+number of GPUs you'd like to use and your training script. Use
+``composer --help`` to see a full list of configurable options.
+
+.. code:: bash
+
+   # run training on 8 GPUs
+   >>> composer -n 8 run_trainer.py
+
+.. seealso::
+
+    Our :doc:`Distributed Training<trainer/distributed_training>` guide and
+    the :mod:`composer.utils.dist` module.
+
+
+DeepSpeed Integration
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer comes with DeepSpeed support, allowing you to leverage their
+full set of features that makes it easier to train large models across
+(1) any type of GPU and (2) multiple nodes. For more details on DeepSpeed,
+see `their website <https://www.deepspeed.ai>`__.
+
+To enable DeepSpeed, simply pass in a config as specified in the
+DeepSpeed docs `here <https://www.deepspeed.ai/docs/config-json/>`__.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu',
+                     deepspeed_config={
```

Comment:
Would it work to pass an empty dictionary into the deepspeed config? Or do fields need to be set?
This would be an api change, but should we allow `deepspeed_config=True` to denote using the default values?

Reply:
Seems like an empty dict should work, but I don't see that tested anywhere. We should add an issue for this.

Reply:
Confirmed to be valid, adding some text to the this section.